Add new bicep function, loadYamlContent

danay1999 opened the pull request:
# Contributing a Pull Request

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, run through the relevant checklist below.

## Contributing to documentation

* [ ] All documentation contributions should be made directly in the [Bicep documentation on Microsoft Docs](https://docs.microsoft.com/azure/azure-resource-manager/bicep/).

## Contributing an example

We are integrating the Bicep examples into the [Azure QuickStart Templates](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md).  If you'd like to contribute new example `.bicep` files that showcase abilities of the language, please follow [these instructions](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md) to add them directly there.  We can still take bug reports and fixes for the existing examples for the time being.

* [ ] This is a bug fix for an existing example
* [ ] I have resolved all warnings and errors shown by the Bicep VS Code extension
* [ ] I have checked that all tests are passing by running `dotnet test`
* [ ] I have consistent casing for all of my identifiers and am using camelCasing unless I have a justification to use another casing style

## Contributing a feature

* [ ] I have opened a new issue for the proposal, or commented on an existing one, and ensured that the Bicep maintainers are good with the design of the feature being implemented
* [ ] I have included "Fixes #{issue_number}" in the PR description, so GitHub can link to the issue and close it when the PR is merged
* [ ] I have appropriate test coverage of my new feature

## Contributing a snippet

* [ ] I have a snippet that is either a single, generic resource or multi resource that uses [parent-child syntax](https://docs.microsoft.com/azure/azure-resource-manager/bicep/child-resource-name-type)
* [ ] I have checked that there is not an equivalent snippet already submitted
* [ ] I have used camelCasing unless I have a justification to use another casing style
* [ ] I have placeholders values that correspond to their property names (e.g. `dnsPrefix: 'dnsPrefix'`), unless it's a property that MUST be changed or parameterized in order to deploy. In that case, I use 'REQUIRED' e.g. [keyData](./src/Bicep.LangServer/Files/SnippetTemplates/res-aks-cluster.bicep#L26)
* [ ] I have my symbolic name as the first tab stop ($1) in the snippet. e.g. [res-aks-cluster.bicep](./src/Bicep.LangServer/Files/SnippetTemplates/res-aks-cluster.bicep)
* [ ] I have a resource name property equal to "name"
* [ ] If applicable, I have set the `location` property to `location: /*${<id>:location}*/'location'` (not `resourceGroup().location`) where `<id>` is a placeholder id, and added `param location string` to the test's main.bicep file so that the resulting main.combined.bicep file used in the tests compiles without errors
* [ ] I have verified that the snippet deploys correctly when used in the context of an actual bicep file

  e.g.

  ```bicep
  resource aksCluster 'Microsoft.ContainerService/managedClusters@2021-03-01' = {
    name: 'name'
  ```

danay1999 committed:
add yaml tests

danay1999 committed:
add yaml tests

PalmEmanuel committed:
New parameter `--force` on Publish command (#9745)

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/Bicep.Cli.E2eTests (#10364)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump peter-evans/create-pull-request from 4 to 5 (#10357)

Bumps [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) from 4 to 5.
- [Release notes](https://github.com/peter-evans/create-pull-request/releases)
- [Commits](https://github.com/peter-evans/create-pull-request/compare/v4...v5)

---
updated-dependencies:
- dependency-name: peter-evans/create-pull-request
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack from 5.77.0 to 5.78.0 in /src/playground (#10376)

Bumps [webpack](https://github.com/webpack/webpack) from 5.77.0 to 5.78.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.77.0...v5.78.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @types/react from 18.0.29 to 18.0.33 in /src/playground (#10371)

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.0.29 to 18.0.33.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump monaco-editor-core from 0.36.1 to 0.37.1 in /src/monarch (#10373)

Bumps [monaco-editor-core](https://github.com/microsoft/vscode) from 0.36.1 to 0.37.1.
- [Release notes](https://github.com/microsoft/vscode/releases)
- [Commits](https://github.com/microsoft/vscode/commits)

---
updated-dependencies:
- dependency-name: monaco-editor-core
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @microsoft/vscode-azext-utils in /src/vscode-bicep (#10379)

Bumps [@microsoft/vscode-azext-utils](https://github.com/Microsoft/vscode-azuretools) from 0.5.1 to 1.1.0.
- [Release notes](https://github.com/Microsoft/vscode-azuretools/releases)
- [Commits](https://github.com/Microsoft/vscode-azuretools/commits)

---
updated-dependencies:
- dependency-name: "@microsoft/vscode-azext-utils"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.56.0 to 5.57.1 in /src/monarch (#10370)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.56.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.0 to 5.57.1 in /src/playground (#10368)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser in /src/vscode-bicep (#10372)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.0 to 5.57.1 in /src/highlightjs (#10369)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack-dev-server from 4.13.1 to 4.13.2 in /src/playground (#10355)

Bumps [webpack-dev-server](https://github.com/webpack/webpack-dev-server) from 4.13.1 to 4.13.2.
- [Release notes](https://github.com/webpack/webpack-dev-server/releases)
- [Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-server/compare/v4.13.1...v4.13.2)

---
updated-dependencies:
- dependency-name: webpack-dev-server
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @microsoft/vscode-azext-azureutils in /src/vscode-bicep (#10359)

Bumps [@microsoft/vscode-azext-azureutils](https://github.com/Microsoft/vscode-azuretools) from 1.0.1 to 1.1.1.
- [Release notes](https://github.com/Microsoft/vscode-azuretools/releases)
- [Commits](https://github.com/Microsoft/vscode-azuretools/commits)

---
updated-dependencies:
- dependency-name: "@microsoft/vscode-azext-azureutils"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump rimraf from 4.4.1 to 5.0.0 in /src/Bicep.Cli.E2eTests (#10385)

Bumps [rimraf](https://github.com/isaacs/rimraf) from 4.4.1 to 5.0.0.
- [Release notes](https://github.com/isaacs/rimraf/releases)
- [Changelog](https://github.com/isaacs/rimraf/blob/main/CHANGELOG.md)
- [Commits](https://github.com/isaacs/rimraf/compare/v4.4.1...v5.0.0)

---
updated-dependencies:
- dependency-name: rimraf
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/highlightjs (#10375)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.0 to 5.57.1 in /src/textmate (#10374)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump rimraf from 4.1.2 to 5.0.0 in /src/vscode-bicep (#10363)

Bumps [rimraf](https://github.com/isaacs/rimraf) from 4.1.2 to 5.0.0.
- [Release notes](https://github.com/isaacs/rimraf/releases)
- [Changelog](https://github.com/isaacs/rimraf/blob/main/CHANGELOG.md)
- [Commits](https://github.com/isaacs/rimraf/compare/v4.1.2...v5.0.0)

---
updated-dependencies:
- dependency-name: rimraf
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/textmate (#10380)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 8.37.0 to 8.38.0 in /src/Bicep.Cli.E2eTests (#10358)

Bumps [eslint](https://github.com/eslint/eslint) from 8.37.0 to 8.38.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.37.0...v8.38.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/vscode-bicep (#10378)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.53.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 8.37.0 to 8.38.0 in /src/monarch (#10367)

Bumps [eslint](https://github.com/eslint/eslint) from 8.37.0 to 8.38.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.37.0...v8.38.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump typescript from 5.0.3 to 5.0.4 in /src/highlightjs (#10362)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.0.3 to 5.0.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.0.3...v5.0.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/monarch (#10361)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser in /src/Bicep.Cli.E2eTests (#10384)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 8.37.0 to 8.38.0 in /src/textmate (#10381)

Bumps [eslint](https://github.com/eslint/eslint) from 8.37.0 to 8.38.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.37.0...v8.38.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump typescript from 5.0.3 to 5.0.4 in /src/textmate (#10382)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.0.3 to 5.0.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.0.3...v5.0.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/playground (#10365)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.56.0 to 5.57.1.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.57.1/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump eslint from 8.37.0 to 8.38.0 in /src/highlightjs (#10366)

Bumps [eslint](https://github.com/eslint/eslint) from 8.37.0 to 8.38.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.37.0...v8.38.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump typescript from 5.0.2 to 5.0.4 in /src/monarch (#10377)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.0.2 to 5.0.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.0.2...v5.0.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump typescript from 5.0.3 to 5.0.4 in /src/Bicep.MSBuild.E2eTests (#10383)

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 5.0.3 to 5.0.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v5.0.3...v5.0.4)

---
updated-dependencies:
- dependency-name: typescript
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

danay1999 committed:
merge

kalbert312 committed:
Attempt to fix e2e test failing. (#10389)

mishrapratikshya committed:
Fix for decompiler to support templatespec (#10329)

* initial commit for decompiler to support templatespec

jmorerice committed:
Revert "Revert "added cidr function, do not merge until arm deployment (#10125)" (#10181)" (#10330)

This reverts commit 2a9216661d77e0b9797e6b95cd8e4f6f0653a491.

majastrz committed:
Update Local TPN Files (#10394)

Signed-off-by: majastrz <majastrz@users.noreply.github.com>
Co-authored-by: majastrz <majastrz@users.noreply.github.com>

dciborow committed:
make yaml a regular yaml file instead of a json

dciborow committed:
break loading yaml and loading json apart

dciborow committed:
remove experimental settings for bicep

dciborow committed:
revert experimental flags

StephenWeatherford committed:
Add MaxAgeInDays config to use-recent-api-versions rule (#10344)

dciborow commented:
```
  Failed ShouldProvideSignatureHelpBetweenFunctionParentheses_LoadFunctions_CRLF [115 ms]
  Error Message:
   Test method Bicep.LangServer.IntegrationTests.SignatureHelpTests.ShouldProvideSignatureHelpBetweenFunctionParentheses threw exception: 
YamlDotNet.Core.SemanticErrorException: While scanning a plain scalar value, found invalid mapping.
  Stack Trace:
      at YamlDotNet.Core.Parser.ParseNode(Boolean isBlock, Boolean isIndentlessSequence)
   at YamlDotNet.Core.Parser.ParseBlockMappingValue()
   at YamlDotNet.Core.Parser.MoveNext()
   at YamlDotNet.Serialization.NodeDeserializers.ScalarNodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.<>c__DisplayClass4_0.<DeserializeValue>b__0(IParser r, Type t)
   at YamlDotNet.Serialization.NodeDeserializers.DictionaryDeserializer.Deserialize(Type tKey, Type tValue, IParser parser, Func`3 nestedObjectDeserializer, IDictionary result)
   at YamlDotNet.Serialization.NodeDeserializers.DictionaryNodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](IParser parser)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](String input)
   at Bicep.Core.Semantics.Namespaces.SystemNamespaceType.ExtractTokenFromObject(String fileContent) in /home/runner/work/bicep/bicep/src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs:line 1177
   at Bicep.Core.Semantics.Namespaces.SystemNamespaceType.LoadYamlContentResultBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, FunctionCallSyntaxBase functionCall, ImmutableArray`1 argumentTypes) in /home/runner/work/bicep/bicep/src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs:line 1126
```

dciborow committed:
update test

dciborow committed:
expanding test case

kalbert312 committed:
Improve compiler detection of invalid runtime value usages. (#10250)

* Remove the index expression check.

* Add test cases for diagnostics against a non-deploy time expression in for body expressions.

* Renames.

* Update the test cases.

* Remove this, merged into other test due to overlap.

* Consider non-string usages as collection access. String usages would really be property based access.

* Try to evaluate string literals up through variables when checking array access index expressions for dtc violations.

* Add test for variable stack.

* Add a baseline for runtime value tests. Pick a select set of permutations from the var for body scenario set for the baseline.

* Remove some diagnostic noise.

* Change the condition to just exclude integer literal syntax. The scope of this code should only be concerned about string-based property access and multiple types of syntax can evaluate to a string literal. Add more test cases.

* Partially revert commit 5498623. Semantic model's get type info should be used instead.

* Use semantic model GetTypeInfo to do index expression type checks to make DTC validation more robust.

* Split the tests so it's more debuggable. Leave TODO about union checking.

* Revert.

* Update baselines.

* Formatting.

* Add string literal checking for all-string unions. Add tests.

Remove the check for integer literal. I think string based property access is the only thing that needs to be checked here. If a resource is resolvable and something else is used, it's going to be a type error, so having the dtc diagnostic and type diagnostic visible at the same time will make it easier for the programmer to figure out what to do next.

* Add tests for nested resource access.

* Fix entire resource cases.

* Replicate the logic changes from direct visitor to indirect visitor. Test progress.

* Fix test cases expectation of indirect usages of param values to expect BCP78 instead of BCP182.

* Fix copy-paste error.. this visitor was using this method instead.

* Reduce duplication.

* Add some more nested resource test variants. Remove redundant checks.

* Action, not Func.

* Fix incorrect diagnostic.

* Refactor to function.

* Add existing resource tests.

* Add interpolated object property key tests.

* Rename the tests. This does not apply to all for body expression contexts.

* PR feedback: Move exhaustive cases to the baselines.

* PR feedback: include the violating property name in the diagnostic. This will allow displaying multiple diagnostics if there's union types involved.

* PR feedback: Reduce the complexity of the scenario tests. Also add the property name to the diagnostic for the indirect case.

* Update baselines.

* Update the json baseline files. Update README.

* Add some expression variants.

* Add a conditional resource collection accessor case.

* Add a test for string array accessor with type any. any() is meant to resolve type errors while DTC is a runtime concept.

* Fix gap of not checking for any or union of ints when checking dtc for whole resource access.

* Fix typo.

danay1999 committed:
merge

danay1999 committed:
merge

danay1999 committed:
merge

dependabot[bot] committed:
Bump vm2 from 3.9.15 to 3.9.16 in /src/vscode-bicep (#10405)

Bumps [vm2](https://github.com/patriksimek/vm2) from 3.9.15 to 3.9.16.
- [Release notes](https://github.com/patriksimek/vm2/releases)
- [Changelog](https://github.com/patriksimek/vm2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/patriksimek/vm2/compare/3.9.15...3.9.16)

---
updated-dependencies:
- dependency-name: vm2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump xml2js and @vscode/vsce in /src/vscode-bicep (#10406)

Bumps [xml2js](https://github.com/Leonidas-from-XIV/node-xml2js) to 0.5.0 and updates ancestor dependency [@vscode/vsce](https://github.com/Microsoft/vsce). These dependencies need to be updated together.


Updates `xml2js` from 0.4.23 to 0.5.0
- [Release notes](https://github.com/Leonidas-from-XIV/node-xml2js/releases)
- [Commits](https://github.com/Leonidas-from-XIV/node-xml2js/commits/0.5.0)

Updates `@vscode/vsce` from 2.18.0 to 2.19.0
- [Release notes](https://github.com/Microsoft/vsce/releases)
- [Commits](https://github.com/Microsoft/vsce/compare/v2.18.0...v2.19.0)

---
updated-dependencies:
- dependency-name: xml2js
  dependency-type: indirect
- dependency-name: "@vscode/vsce"
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

danay1999 committed:
merge

danay1999 committed:
merged

danay1999 committed:
merged

dependabot[bot] committed:
Bump JsonPath.Net from 0.2.1 to 0.4.0 (#10065)

* Bump JsonPath.Net from 0.2.1 to 0.4.0

Bumps [JsonPath.Net](https://github.com/gregsdennis/json-everything) from 0.2.1 to 0.4.0.
- [Release notes](https://github.com/gregsdennis/json-everything/releases)
- [Commits](https://github.com/gregsdennis/json-everything/compare/path-v0.2.1...path-v0.4.0)

---
updated-dependencies:
- dependency-name: JsonPath.Net
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update dotnet lockfiles

* Fix an error due to an API breaking change

* Bump versions for JSON packages

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>
Co-authored-by: Shenglong Li <shenglol@microsoft.com>

danay1999 committed:
merged

shenglol committed:
Formatter v2 part 3: Get static token texts with a switch expression (#10408)

* Add SyntaxFacts.GetText(TokenType)

* Use AsSpan

jabbera committed:
Resource Type parameters should generate BCP120 (#10401)

Co-authored-by: Mike Barry <michael.barry@gmo.com>

danay1999 committed:
merged

jeskew committed:
Use decorators to determine output type (#10418)

* Apply type refinements to outputs

* Expand scenario test to ensure diagnostic is raised w/o constraints on output

Usman0111 committed:
fixed help (#10412)

dciborow commented:
[](url)

anthony-c-martin committed:
Fix for unhandled params file exception (#10423)

danay1999 committed:
merged

dciborow committed:
adding integration tests to LoadFunctionTests

dciborow committed:
Update src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

dciborow committed:
adding yaml specific parsing error

dciborow committed:
code review comments

dciborow committed:
Update src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

Co-authored-by: Anthony Martin <38542602+anthony-c-martin@users.noreply.github.com>

dciborow committed:
comment out additional test

dciborow committed:
add missing ARM_YAML_JSON which is the value of string(loadYamlFunction('test.yml'))

dciborow committed:
adding IObjectParser and derived classes

dciborow committed:
simplify

dciborow committed:
update tests

dciborow committed:
update tests

dciborow committed:
update tests

dciborow committed:
minor fixes

dciborow committed:
fix up interface

dciborow committed:
use common method for both types of loading

dciborow committed:
use common method for both types of loading

dciborow committed:
use common method for both types of loading

dciborow committed:
use common method for both types of loading

dciborow committed:
use common method for both types of loading

dciborow committed:
use common method for both types of loading

dciborow committed:
final refactors

majastrz committed:
added test (#10431)

dependabot[bot] committed:
Bump @typescript-eslint/parser in /src/Bicep.Cli.E2eTests (#10438)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack-dev-server from 4.13.2 to 4.13.3 in /src/playground (#10434)

Bumps [webpack-dev-server](https://github.com/webpack/webpack-dev-server) from 4.13.2 to 4.13.3.
- [Release notes](https://github.com/webpack/webpack-dev-server/releases)
- [Changelog](https://github.com/webpack/webpack-dev-server/blob/master/CHANGELOG.md)
- [Commits](https://github.com/webpack/webpack-dev-server/compare/v4.13.2...v4.13.3)

---
updated-dependencies:
- dependency-name: webpack-dev-server
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @types/react from 18.0.32 to 18.0.35 in /src/vscode-bicep (#10437)

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.0.32 to 18.0.35.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @microsoft/vscode-azext-utils in /src/vscode-bicep (#10444)

Bumps [@microsoft/vscode-azext-utils](https://github.com/Microsoft/vscode-azuretools) from 1.1.0 to 1.2.0.
- [Release notes](https://github.com/Microsoft/vscode-azuretools/releases)
- [Commits](https://github.com/Microsoft/vscode-azuretools/commits)

---
updated-dependencies:
- dependency-name: "@microsoft/vscode-azext-utils"
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump react-bootstrap from 2.7.2 to 2.7.4 in /src/playground (#10439)

Bumps [react-bootstrap](https://github.com/react-bootstrap/react-bootstrap) from 2.7.2 to 2.7.4.
- [Release notes](https://github.com/react-bootstrap/react-bootstrap/releases)
- [Changelog](https://github.com/react-bootstrap/react-bootstrap/blob/master/CHANGELOG.md)
- [Commits](https://github.com/react-bootstrap/react-bootstrap/compare/v2.7.2...v2.7.4)

---
updated-dependencies:
- dependency-name: react-bootstrap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/monarch (#10441)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/textmate (#10440)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.1 to 5.58.0 in /src/highlightjs (#10442)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/vscode-bicep (#10451)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.1 to 5.58.0 in /src/playground (#10443)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.1 to 5.58.0 in /src/monarch (#10445)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/highlightjs (#10450)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump semver from 7.3.8 to 7.4.0 in /src/vscode-bicep (#10448)

Bumps [semver](https://github.com/npm/node-semver) from 7.3.8 to 7.4.0.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/main/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v7.3.8...v7.4.0)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/parser from 5.57.1 to 5.58.0 in /src/textmate (#10447)

Bumps [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/parser) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/parser/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/parser)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/parser"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @typescript-eslint/eslint-plugin in /src/Bicep.Cli.E2eTests (#10453)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.57.1 to 5.58.0.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.58.0/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

danay1999 committed:
merged

danay1999 committed:
merged

dependabot[bot] committed:
Bump eslint-config-prettier from 8.7.0 to 8.8.0 in /src/vscode-bicep (#10452)

Bumps [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) from 8.7.0 to 8.8.0.
- [Release notes](https://github.com/prettier/eslint-config-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-config-prettier/compare/v8.7.0...v8.8.0)

---
updated-dependencies:
- dependency-name: eslint-config-prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump System.IO.Abstractions.TestingHelpers from 19.1.18 to 19.2.11 (#10435)

* Bump System.IO.Abstractions.TestingHelpers from 19.1.18 to 19.2.11

Bumps [System.IO.Abstractions.TestingHelpers](https://github.com/TestableIO/System.IO.Abstractions) from 19.1.18 to 19.2.11.
- [Release notes](https://github.com/TestableIO/System.IO.Abstractions/releases)
- [Commits](https://github.com/TestableIO/System.IO.Abstractions/compare/v19.1.18...v19.2.11)

---
updated-dependencies:
- dependency-name: System.IO.Abstractions.TestingHelpers
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix dotnet lockfiles

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anthony Martin <38542602+anthony-c-martin@users.noreply.github.com>

dependabot[bot] committed:
Bump @microsoft/applicationinsights-web from 2.8.11 to 3.0.0 in /src/playground (#10446)

* Bump @microsoft/applicationinsights-web in /src/playground

Bumps [@microsoft/applicationinsights-web](https://github.com/microsoft/ApplicationInsights-JS) from 2.8.11 to 3.0.0.
- [Release notes](https://github.com/microsoft/ApplicationInsights-JS/releases)
- [Changelog](https://github.com/microsoft/ApplicationInsights-JS/blob/main/RELEASES.md)
- [Commits](https://github.com/microsoft/ApplicationInsights-JS/compare/2.8.11...3.0.0)

---
updated-dependencies:
- dependency-name: "@microsoft/applicationinsights-web"
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix build

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anthony Martin <38542602+anthony-c-martin@users.noreply.github.com>

dependabot[bot] committed:
Bump @types/react from 18.0.33 to 18.0.35 in /src/playground (#10449)

Bumps [@types/react](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react) from 18.0.33 to 18.0.35.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react)

---
updated-dependencies:
- dependency-name: "@types/react"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump System.IO.Abstractions from 19.2.4 to 19.2.11 (#10433)

* Bump System.IO.Abstractions from 19.2.4 to 19.2.11

Bumps [System.IO.Abstractions](https://github.com/TestableIO/System.IO.Abstractions) from 19.2.4 to 19.2.11.
- [Release notes](https://github.com/TestableIO/System.IO.Abstractions/releases)
- [Commits](https://github.com/TestableIO/System.IO.Abstractions/compare/v19.2.4...v19.2.11)

---
updated-dependencies:
- dependency-name: System.IO.Abstractions
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix dotnet lockfiles

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anthony Martin <38542602+anthony-c-martin@users.noreply.github.com>

danay1999 committed:
merged

dciborow committed:
nuget sync

dciborow committed:
restore packages.lock.json

dciborow committed:
restore csproj

danay1999 committed:
added sharpyaml

danay1999 committed:
refactorings

dciborow committed:
restore deps

danay1999 committed:
fixed test cases

danay1999 committed:
added completion tests

dciborow committed:
remove extra abstract

dciborow committed:
remove extra imports

danay1999 committed:
review comments

danay1999 committed:
review comments

dciborow committed:
updating BRM generate for user-defined types, and add integration tes… (#10390)

* updating BRM generate for user-defined types, and add integration test for experimental settings

* remove duplicate code

* removed to many outputs

* add support for description, and all param to override type description

* revert unneeded changes

* fix pointer to description

* fix pointer to description

* fix pointer to description

* fix pointer to description

* support nested user-defined types

* update style

* cleanup code

* update style

* change from use of lazy variable

* start to use nuget package

* back to working with extras commented out

* try to use nuget tools

* parameters are working

* all unit tests working

* cleanup test

* cleanup

* cleanup

* more cleanup

* Update src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

Co-authored-by: Shenglong Li <shenglong.li@outlook.com>

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

* Apply suggestions from code review

* Update ArmTemplateSemanticModel.cs

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

* Update src/Bicep.RegistryModuleTool.TestFixtures/SampleFiles/Experimental/main.bicep

* Update src/Bicep.RegistryModuleTool.TestFixtures/SampleFiles/Experimental/main.bicep

* Update MainArmTemplateFile.cs

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

* Update MainArmTemplateFile.cs

* Update MainArmTemplateFile.cs

* Update src/Bicep.RegistryModuleTool/ModuleFiles/MainArmTemplateFile.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

---------

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>
Co-authored-by: Shenglong Li <shenglong.li@outlook.com>

StephenWeatherford committed:
Temporary fix for VS build (#10485)

Co-authored-by: Stephen Weatherford <Stephen.Weatherford.com>

dependabot[bot] committed:
Bump Microsoft.Extensions.Configuration.Binder from 7.0.3 to 7.0.4 (#10360)

* Bump Microsoft.Extensions.Configuration.Binder from 7.0.3 to 7.0.4

Bumps [Microsoft.Extensions.Configuration.Binder](https://github.com/dotnet/runtime) from 7.0.3 to 7.0.4.
- [Release notes](https://github.com/dotnet/runtime/releases)
- [Commits](https://github.com/dotnet/runtime/compare/v7.0.3...v7.0.4)

---
updated-dependencies:
- dependency-name: Microsoft.Extensions.Configuration.Binder
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Fix dotnet lockfiles

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Anthony Martin <38542602+anthony-c-martin@users.noreply.github.com>

danay1999 committed:
changed jsonPath to pathFilter

dependabot[bot] committed:
Bump vm2 from 3.9.16 to 3.9.17 in /src/vscode-bicep (#10482)

Bumps [vm2](https://github.com/patriksimek/vm2) from 3.9.16 to 3.9.17.
- [Release notes](https://github.com/patriksimek/vm2/releases)
- [Changelog](https://github.com/patriksimek/vm2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/patriksimek/vm2/compare/3.9.16...3.9.17)

---
updated-dependencies:
- dependency-name: vm2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

StephenWeatherford committed:
Fix 9451 Goto def not working with custom registry aliasess (#10476)

StephenWeatherford committed:
remove "experimental" from bicep.containerRegistryCompletions.useAllAccessible setting (#10457)

* Clean up code and comments

* use cancellationtoken

* https://github.com/Azure/bicep/issues/10456

* Remove "experimental" from setting name

* spelling

* spelling

* Return async enumerable from ACS query

* CR comment fixes

* CR comment fixes

anamikapan committed:
semantic token addition (#10472)

danay1999 committed:
Merge branch 'main' into danay1999/yaml

dciborow committed:
Update src/Bicep.Core/Semantics/ObjectParser.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

dciborow committed:
Update src/Bicep.Core/Semantics/JsonObjectParser.cs

Co-authored-by: Jonny Eskew <jeskew@users.noreply.github.com>

dciborow committed:
Update JsonObjectParser.cs
